primitive/src/main/java/io/atomix/primitive/service/impl/DefaultBackupInput.java

```diff
@@ -91,6 +91,12 @@ public BackupInput read(Buffer buffer) {
     return this;
   }
 
+  @Override
+  public BackupInput read(final ByteBuffer buffer) {
```

Comment:
Did we think about using DirectBuffer from agrona?

Reply:
I'd like to tackle change the serialization at some point, but I feel like that's a bigger issue :sweat_smile:

Reply:
ok just thought this would be easier to integrate with zeebe but lets see.